lib/spack/spack/cmd/__init__.py

```diff
@@ -219,7 +220,13 @@ def get_arg(name, default=None):
         hlen = None
 
     nfmt = '.' if namespace else '_'
-    ffmt = '$%+' if flags else ''
+    ffmt = ''
+    if full_compiler or flags:
```

Comment:
I think the ```--show-flags``` option should be removed and when you specify ```-F``` you get the version and the flags. At the moment IMO it's confusing when ```-F``` doesn't print the flags too and you also need ```-f```. If the option to include the compiler version was ```--show-compiler-version``` it would be fine.